# 1. 데이터 확인

In [35]:
import pandas as pd
import numpy as np
# from matplotlib import pyplot as plt
# import seaborn as sns

## 1-1. 데이터 가져오기

In [3]:
wanted_nps_df = pd.read_csv("../company_nps_data.csv", encoding='cp949')

In [4]:
wanted_nps_df.head()

회사ID      연매출액    년도   월  월별_연금보험료  월별_직원수
0  233757  41688077  2018   7   8912380      36
1  123464   1760695  2018   3    942380       5
2  342124   3221341  2018   7    781180       4
3  386420   4815584  2015  11   3795900      14
4  129243   9799736  2018  10  40724680     151

In [5]:
wanted_nps_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99801 entries, 0 to 99800
Data columns (total 6 columns):
 #   Column    Non-Null Count  Dtype
---  ------    --------------  -----
 0   회사ID      99801 non-null  int64
 1   연매출액      99801 non-null  int64
 2   년도        99801 non-null  int64
 3   월         99801 non-null  int64
 4   월별_연금보험료  99801 non-null  int64
 5   월별_직원수    99801 non-null  int64
dtypes: int64(6)
memory usage: 4.6 MB


In [6]:
wanted_nps_df.shape

(99801, 6)

## 1-2. 연도

In [7]:
year = list(wanted_nps_df.년도.unique())
year = sorted(year)
year = list(map(str, year))
print(year)

['2015', '2016', '2017', '2018', '2019']


## 1-3. 연매출액

In [21]:
sales = list(wanted_nps_df.연매출액)
sales_mean = pd.Series(sales)
sales_mean = sales_mean.mean()
print(sales_mean)

30535893.434244145


In [24]:
sales_max = pd.Series(sales)
sales_max = sales_max.max()
print(sales_max)

11328779396


In [25]:
sales_min = pd.Series(sales)
sales_min = sales_min.min()
print(sales_min)

-7882176


## 1-4. 월별_연금보험료

In [8]:
print((wanted_nps_df['월별_연금보험료'] == 0).values.sum())

13


☑️ 월별_연금보험료가 `0`원인 경우가 `13`건 포함되어 있다.

In [17]:
premium = list(wanted_nps_df.월별_연금보험료)
# print(len(premium))
for i in premium:
    if i == 0:
        premium.remove(0)

In [18]:
premium_mean = pd.Series(premium)
premium_mean = premium_mean.mean()
print(premium_mean)

9735347.33274542


In [19]:
premium_max = pd.Series(premium)
premium_max = premium_max.max()
print(premium_max)

978067260


In [26]:
premium_min = pd.Series(sales)
premium_min = premium_min.min()
print(premium_min)

-7882176


## 1-5. 월별_직원수

In [10]:
print((wanted_nps_df['월별_직원수'] == 0).values.sum())

13


☑️ 월별_직원수가 `0`명인 경우가 `13`건 포함되어 있다.

In [14]:
employee = list(wanted_nps_df.월별_직원수)
# print(len(employee))
for i in employee:
    if i == 0:
        employee.remove(0)

In [15]:
employee_mean = pd.Series(employee)
employee_mean = employee_mean.mean()
print(employee_mean)

37.62833206397563


In [16]:
employee_max = pd.Series(employee)
employee_max = employee_max.max()
print(employee_max)

3133


In [27]:
employee_min = pd.Series(employee)
employee_min = employee_min.min()
print(employee_min)

3


## 1-6. 회사ID

In [12]:
comp_id = list(wanted_nps_df.회사ID.unique())
comp_id = sorted(comp_id)

# comp_id = pd.Series(wanted_nps_df.회사ID.unique())
# comp_id = comp_id.sort_values()
# display(comp_id)
print(len(comp_id))

5000


In [13]:
wanted_nps_df['회사ID'].value_counts()

# len(set(wanted_nps_df['회사ID']))

155754     50
237625     50
296819     50
395238     50
442525     50
           ..
2256869     1
573311      1
317847      1
440452      1
1983879     1
Name: 회사ID, Length: 5000, dtype: int64

In [34]:
is_zero = (wanted_nps_df['월별_연금보험료'] == 0)
zero = wanted_nps_df[is_zero]
zero
zero_list = list(zero.회사ID.unique())
print(len(zero_list))

회사ID      연매출액    년도   월  월별_연금보험료  월별_직원수
4450   478999    394790  2016   3         0       0
6162   568445    740245  2018  10         0       0
7251   288014   3709457  2016  11         0       0
12176  568445    740245  2018   8         0       0
20070  550603    416881  2018   1         0       0
33869  439536    237277  2016  10         0       0
39671  602137    892176  2018   6         0       0
49592  291968   4059302  2016   4         0       0
54026  277054     95478  2016   5         0       0
64139  472629   1915700  2018   1         0       0
69239  157237  27020003  2017   5         0       0
82809  217292    245945  2016   6         0       0
93481  288014   3709457  2016   1         0       0

☑️ 월별_연금보험료가 `0`원인 경우를 확인하면서,<br>
　　동시에 월별_직원수가 `0`원인 것을 알 수 있었다.<br>
　　그 case는 총 `13`가지 였고,<br>
　　중복되었던 것을 고려한다면 `11`가지가 된다.<br>

☑️ 회사ID는 중복없이 `4988`개이다.<br>

### 💡 check!
💠 연도는 `2015`, `2016`, `2017`, `2018`, `2019` 까지 나타난다.<br>
💠 월별_연금보험료가 0원이면서,<br>
　　월별_직원수가 0인 데이터가 있는 것을 확인했으니, 고려하자.<br>
💠 회사ID column 에는 중복없이 회사ID 가 나열되어있는 것이 아니다.<br>
　　(최대 50개월의 정보를 가지고 있는 경우)<br>

### 📝 summary
#### 💠 연매출액
- 평균 : 30535893천원 (₩ 30,535,893,000)<br>
- 최대값 : 11328779396천원 (₩ 11,328,779,396,000)<br>
- 최솟값 : -7882176천원 (₩ -7,882,176,000)<br>

#### 💠 월별_연금보험료
- 평균 : 9735347 (₩ 9,735,347)<br>
- 최대값 : 978067260 (₩ 978,067,260)<br>
- 최솟값 : -7882176 (₩ -7882176)<br>

#### 💠 월별_직원수
- 평균 : 37명<br>
- 최대값 : 3133명<br>
- 최솟값 : 3<br>